gaeguli/pipeline.c

```diff
@@ -124,9 +126,11 @@ typedef enum
   PROP_SOURCE = 1,
   PROP_DEVICE,
   PROP_ENCODING_METHOD,
+  PROP_SOURCE_PIPELINE,
+  PROP_ENCODING_PIPELINE,
 
   /*< private > */
-  PROP_LAST = PROP_ENCODING_METHOD,
+  PROP_LAST = PROP_ENCODING_PIPELINE,
 } _GaeguliPipelineProperty;
 
 static GParamSpec *properties[PROP_LAST + 1];
```

Comment:
Perhaps `PROP_LAST` could have a value of its own and not be an alias for the last actual property identifier in the enumeration. You would avoid the need to modify it each time you add a new property (hence eliminate a potential place for making an error) and could use it directly when declaring `properties` array:

```static GParamSpec *properties[PROP_LAST];``` (without +1).

Reply:
Thank you for good advice.